software/python/robot.py

```diff
@@ -0,0 +1,79 @@
+import serial
+import serial.tools.list_ports
+import struct
+import numpy as np
+
+class Robot:
+    def __init__(self):
+        super().__init__()
+        self.speed = 1
+        self.motors = [
+            {"r":38, "R":98, "angle": 1/6*np.pi},
+            {"r":38, "R":98, "angle": 11/6*np.pi},
+            {"r":22, "R":98, "angle": np.pi    }
+        ]
+
+        port = [i[0] for i in serial.tools.list_ports.comports()][0]
+        self.ser = serial.Serial(port)
+
+    def setSpeed(self, speed):
+        self.speed = speed
+
+
+    def stop(self):
+        self.move(0, 0, 0)
+
+    # self.speed - from 0 to 1, float, out of maximum speed
+    # 32767 on max valid
+    # -32768 on min valid
+
+
+    # portion - turning / driving straight: 1 - spin // 0 - straight
+    
+    def move_left(self): # correct
+        self.move_omni(np.pi/2, 1000, 0, self.motors)
+    def move_right(self): # correct
+        self.move_omni(3*np.pi/2, 1000, 0, self.motors)    
+    def move_backward(self): # 
+        self.move_omni(np.pi, 1000, 0)
+    def move_forward(self): # correct
+        self.move_omni(0, 1000, 0)
+    def spin_right(self, speed=5): # correct
```

Comment:
Why not have one method that takes either positive or negative values? Why have this level of abstraction?